tableschema_sql/mapper.py

```diff
@@ -75,10 +103,20 @@ def convert_descriptor(self, bucket, descriptor, index_fields=[], autoincrement=
                         checks.append(Check('"%s" REGEXP \'%s\'' % (field.name, value)))
                 elif name == 'enum':
                     column_type = sa.Enum(*value, name='%s_%s_enum' % (table_name, field.name))
-            column = sa.Column(*([field.name, column_type] + checks),
-                nullable=nullable, comment=comment, unique=unique)
-            columns.append(column)
-            column_mapping[field.name] = column
+
+            # defines the column
+            if self.has_key_value(descriptor, field.name, 'protected'):
```

Comment:
Pretty sure with this function call this comes out to O(n^2).

You have access to the field right here so you can just do,

```python
if 'protected' in field:
```